### PR TITLE
[fix](ldap) fix ldap related errors

### PR DESCRIPTION
**Pull request by luozenglin**
1. fix ldap user show grants return null pointer exception;
2. fix ldap user show databases return no authority db;
3. ldap authentication supports catalog level;

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luozenglin** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.31 seconds
 stream load tsv:          419 seconds loaded 74807831229 Bytes, about 170 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          80.1 seconds inserted 10000000 Rows, about 124K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230523144119_clickbench_pr_149134.html

**Comment by luozenglin** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.35 seconds
 stream load tsv:          443 seconds loaded 74807831229 Bytes, about 161 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          77.6 seconds inserted 10000000 Rows, about 128K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230524070535_clickbench_pr_149564.html

**Comment by luozenglin** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.48 seconds
 stream load tsv:          445 seconds loaded 74807831229 Bytes, about 160 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          26.2 seconds inserted 10000000 Rows, about 381K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230524113249_clickbench_pr_149782.html

File: fe/fe-core/src/main/java/org/apache/doris/ldap/LdapAuthenticate.java [hunk 1]
```diff
@@ -64,11 +66,15 @@ public static boolean authenticate(ConnectContext context, String password, Stri
...
-        if (userIdentity == null) {
+        List<UserIdentity> userIdentities = Env.getCurrentEnv().getAuth()
+                .getUserIdentityForLdap(qualifiedUser, remoteIp);
+        UserIdentity userIdentity;
+        if (userIdentities.isEmpty()) {
             userIdentity = tempUserIdentity;
             LOG.debug("User:{} does not exists in doris, login as temporary users.", userName);
             context.setIsTempUser(true);
+        } else {
+            userIdentity = userIdentities.get(0);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/ldap/LdapAuthenticate.java:77]: Why using the first one? Or is it just an arbitrary selection?

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**